bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Control.java

```diff
@@ -3969,9 +3969,14 @@ void subclass () {
  */
 public Point toControl (int x, int y) {
 	checkWidget ();
-	Point displayPointInPixels = getDisplay().translateLocationInPixelsInDisplayCoordinateSystem(x, y);
-	final Point controlPointInPixels = toControlInPixels(displayPointInPixels.x, displayPointInPixels.y);
-	return DPIUtil.scaleDown(controlPointInPixels, getZoom());
+	int zoom = getZoom();
+	if (getDisplay().isRescalingAtRuntime()) {
+		Point displayPointInPixels = getDisplay().translateLocationInPixelsInDisplayCoordinateSystem(x, y);
+		final Point controlPointInPixels = toControlInPixels(displayPointInPixels.x, displayPointInPixels.y);
+		return DPIUtil.scaleDown(controlPointInPixels, zoom);
+	} else {
+		return DPIUtil.scaleDown(toControlInPixels(DPIUtil.scaleUp(x, zoom), DPIUtil.scaleUp(y, zoom)), zoom);
```

Comment:
Move outside of `else`

Reply:
I personally find it more readable inside if-else, because both scenarios are completely distinct branches. But it makes it more obvious, that the additional code was only added and is activated in the rescale scenario, so I adapted all places accordingly.

Reply:
Thank you